src/main/java/io/github/thebusybiscuit/slimefun4/core/guide/SlimefunGuideUnlockMode.java

```diff
@@ -0,0 +1,99 @@
+package io.github.thebusybiscuit.slimefun4.core.guide;
+
+import io.github.thebusybiscuit.slimefun4.api.researches.Research;
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+import io.github.thebusybiscuit.slimefun4.integrations.VaultIntegration;
+import java.util.Locale;
+import javax.annotation.Nonnull;
+import org.bukkit.entity.Player;
+
+/**
+ * This enum holds the different unlock research modes a {@link SlimefunGuide} can have.
+ * Each constant corresponds to a research unlock mode and a unlock provider
+ *
+ * @author StarWishsama
+ * @see SlimefunGuide
+ * @see SlimefunGuideImplementation
+ * @see SlimefunGuideUnlockProvider
+ */
+public enum SlimefunGuideUnlockMode {

```

Comment:
I would wonder, perhaps, if having this as an interface rather than an enum would be better? It would mean addons could create custom open types with ease which I could see being very powerful with little loss from the lack of an enum?

Reply:
I only thought unlock a research in vanilla sf before, add a unlock provider variable for slimefun item for addon seems more usable.

Reply:
To expand on what sefi said you could even provide methods for getting the item when a research isn't unlocked, that would allow for even more flexibility

Reply:
I don't get you about __unlocked item__, any example?

Reply:
> I don't get you about _locked research item_, any example?

Currently in slimefun if an item isnt researched it shows up in the guide as a barrier with a name & the experience cost in it. So adding a method to create that item

Reply:
Implemented, thanks for advice.